http/handler.go

```diff
@@ -303,6 +304,21 @@ func wrapGenericHandler(core *vault.Core, h http.Handler, props *vault.HandlerPr
 		r = r.WithContext(ctx)
 		r = r.WithContext(namespace.ContextWithNamespace(r.Context(), namespace.RootNamespace))
 
+		// Set some response headers with raft node id (if applicable) and hostname, if available
+		nodeID := core.GetRaftNodeID()
+		if nodeID != "" {
+			w.Header().Set("X-Vault-Raft-Node-ID", nodeID)
+		}
+		info, err := hostutil.CollectHostInfo(ctx)
+
+		// These headers aren't essential, so don't return an error here - just log it
+		if err != nil {
+			core.Logger().Error("error retrieving host info", "err", err)
```

Comment:
Why not https://golang.org/pkg/os/#Hostname ?

Reply:
I wasn't aware that existed and I saw other code inside Vault using `hostutil`. Is `os.Hostname` preferred?

Reply:
hostutil is calling os.Hostname, and doing a bunch of other stuff besides.  I think in the request hot path we should do as little as possible.

Reply:
Sounds good, I'll update it.